src/protocols/gwmp/hpr_gwmp_worker.erl

```diff
@@ -260,7 +262,7 @@ packet_up_to_push_data(Up, GatewayTime, GatewayLocation) ->
         MAC,
         #{
             time => iso8601:format(
-                calendar:system_time_to_universal_time(GatewayTime, millisecond)
+                calendar:system_time_to_universal_time(Timestamp, millisecond)
```

Comment:
The timestamp can contain milliseconds so you don't always have to read into metadata for better time resolution.

```erl
%% calendar strips milliseconds, add them back for formatting.
{Date, {Hour, Min, Sec0}} = calendar:system_time_to_universal_time(GatewayTime, millisecond),
Sec = Sec0 + (GatewayTime rem 1000 / 1000),
Time = iso8601:format({Date, {Hour, Min, Sec}}),
```